source/common/router/router.cc

```diff
@@ -596,15 +597,27 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   return Http::FilterHeadersStatus::StopIteration;
 }
 
+// TODO(alyssawilk) create from cluster config.
```

Comment:
How could a `Router::GenericConnPool` created from cluster config?
The cluster doesn't even know if a filter requires `Router::GenericConnPool`, `TcpProxy::GenericConnPool`, or a `WeirdConnPool`.

My understanding the GenericConnPool is generic to the impl of ConnPool, but there is no such a `generic` conn pool interface for HttpConnManager, TcpProxy and Redis. Each above protocol has its individual `GenericConnPool`.
It might be more straighforward if you rename `GenericConnPool` to `RouterGenericConnPool` in below line

Reply:
Did the factory todo - take a look and let me know if you have further questions